public-ui/src/app/data-browser/views/pm/pm-react.component.tsx

```diff
@@ -0,0 +1,488 @@
+                this.setState({loading: false});
+            }
+          }
+          if (searchText) {
+            this.setState({selectedGroup: PMGroups.filter(conceptgroup =>
+              conceptgroup.groupName.toLowerCase().
+              includes(searchText.toLowerCase()))[0]});
+          } else {
+            this.setState({selectedGroup: PMGroups[0]});
+          }
+          this.setUnit(PMGroups[0].concepts[0]);
+    });
+  }
+
+  getPMCountData() {
+    api.getCountAnalysis('Physical Measurements', 'pm').then(
+        (result) => {
+            this.setState({domainCountAnalysis: result});
+    });
+  }
+
+  organizeGenders(concept: any) {
+    const analysis = concept.analyses.genderAnalysis;
+    let male = null;
+    let female = null;
+    let other = null;
+
+    // No need to do anything if only one gender
+    if (analysis.results.length <= 1) {
+      return;
+    }
+    const results = [];
+    for (const g of analysis.results) {
+      if (g.stratum2 === '8507') {
+        male = g;
+      } else if (g.stratum2 === '8532') {
+        female = g;
+      } else if (g.stratum2 === '0') {
+        other = g;
+      }
+    }
+
+    // Order genders how we want to display  Male, Female , Others
+    if (male) { results.push(male); }
+    if (female) { results.push(female); }
+    if (other) { results.push(other); }
+    analysis.results = results;
+  }
+
+  setAnalysisStratum(results: any) {
+    for (const r of results) {
+      if (r.analysisStratumName === null) {
+        r.analysisStratumName = GENDER_STRATUM_MAP[r.stratum3];
+      }
+    }
+  }
+
+  arrangeConceptAnalyses(concept: any) {
+    if (concept.analyses.genderAnalysis) {
+      this.organizeGenders(concept);
+    }
+
+    let genders = ['8507', '8532', '0'];
+    let prevResult;
+    for (const gca of concept.analyses.measurementGenderCountAnalysis) {
+      if (gca.results.length < 3) {
+        for (const result of gca.results) {
+          prevResult = result;
+          genders = genders.filter(item => item !== result.stratum3);
+        }
+        for (const gender of genders) {
+          const missingResult = { ...prevResult };
+          missingResult.stratum3 = gender;
+          missingResult.countValue = 20;
+          missingResult.sourceCountValue = 20;
+          gca.results.push(missingResult);
+        }
+      }
+      this.setAnalysisStratum(gca.results);
+    }
+  }
+
+  showMeasurement(pmConceptGroup: any, concept: any) {
+    this.setUnit(concept);
+    this.setState({selectedGroup: pmConceptGroup, selectedConcept: concept});
+  }
+
+  setUnit(concept: any) {
+    const unitNames = [];
+    if (concept.analyses && concept.analyses.measurementGenderCountAnalysis) {
+      for (const r of concept.analyses.measurementGenderCountAnalysis) {
+        let tempUnitNames = r.results.map(({ stratum2 }) => stratum2);
+        tempUnitNames = tempUnitNames.filter((elem, index, self) => index === self.indexOf(elem));
```

Comment:
Nit: can combine to a single line:
```suggestion
        const tempUnitNames = r.results.map(({ stratum2 }) => stratum2).filter((elem, index, self) => index === self.indexOf(elem));
```

Reply:
Thanks. changed this.